index.bs

```diff
@@ -3499,6 +3532,87 @@ To <dfn noexport>set an OS-support header</dfn> given a [=header list=]
 1. [=header list/Set a structured field value=] given
     ("<code>[=Attribution-Reporting-Support=]</code>", |dict|) in |headers|.
 
+
+# Report Verification Algorithms # {#report-verification}
+
+"<dfn><code>Sec-Attribution-Reporting-Private-State-Token</code></dfn>" is a [=structured header=] used
+to orchestrate report verification.
+
+* When [=setting trigger verification request headers=], the field is a collection of unsigned masked tokens formatted as a [=structured header/list=] of [=structured header/string=].
+* When [=receiving trigger verification tokens=], the field is a collection of signed masked tokens formatted as a [=structured header/list=] of [=structured header/string=].
+* When [=generating attribution report headers=], the header field is a single signed unmasked token formatted as a [=structured header/string=].
+
+<h3 id="initiate-trigger-verification">Initiate trigger verification</h3>
+
+To <dfn noexport>set trigger verification request headers</dfn> given a [=request=] |request|
+and a [=suitable origin=] |destination|:
+
+1. Let |issuer| be |request|'s [=request/URL=]'s [=url/origin=].
+1. If |issuer| is not [=check if an origin is suitable|suitable=], return.
+1. Let (|issuerKeys|, |pretokens|, |cryptoProtocolVersion|) be the result of [=looking up the key commitments=] for |issuer|.
+1. If |issuerKeys| is null, return.
+1. Let |ids| be a new [=list=].
+1. Let |base64EncodedMaskedMessages| be a new [=list=].
+1. While |ids|'s [=list/size=] is less than [=verification tokens per trigger=]:
+    1. Let |id| be the result of [=generating a random UUID=].
+    1. [=list/Append=] |id| to |ids|.
+    1. Let |message| be the [=string/concatenate|concatenation=] of |id| and |destination|.
+    1. Let |maskedMessage| be the result of [=generating masked tokens=] with |issuerKeys| and |message|.
+    1. Let |base64EncodedMaskedMessage| be the [=forgiving-base64 encoding=] of |maskedMessage|.
+    1. Let |fieldValue| be the result of [=structured header/serializing a string=] with |base64EncodedMaskedMessage|.
+    1. [=list/Append=] |fieldValue| to |base64EncodedMaskedMessages|.
+1. Let |request|'s [=request/trigger verification metadata=] be a [=trigger verification metadata=] with the items:
+    : [=trigger verification metadata/pretokens=]
+    :: |pretokens|
+    : [=trigger verification metadata/ids=]
+    :: |ids|
+1. [=header list/Set a structured field value=] given
+    ("<code>[=Sec-Attribution-Reporting-Private-State-Token=]</code>", the result of [=structured header/serializing a list=] with |base64EncodedMaskedMessages|)
```

Comment:
Doesn't https://fetch.spec.whatwg.org/#concept-header-list-set-structured-header already handle the serialization of the list for us? Same below.

Reply:
You're right, updated!